src/v1/combo-chart/combo-chart.ts

```diff
@@ -0,0 +1,669 @@
+/**
+ * @file Combo Chart Implementation from chart.js library
+ *
+ * @fileoverview
+ *
+ * @author Rohit Singh <rohit.singh@thoughtspot.com>
+ *
+ * Copyright: ThoughtSpot Inc. 2023
+ */
+
+import {
+  ChartColumn,
+  ChartConfig,
+  ChartModel,
+  ChartToTSEvent,
+  ColumnType,
+  CustomChartContext,
+  DataPointsArray,
+  getChartContext,
+  PointVal,
+  Query,
+  ValidationResponse,
+  VisualProps,
+} from "@thoughtspot/ts-chart-sdk";
+import Chart from "chart.js/auto";
+import { htmlLegendPlugin } from "./legend-plugin";
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import _, { find } from "lodash";
+
+const logger = console;
+
+Chart.register(ChartDataLabels);
+
+let globalChartReference: Chart;
+
+const visualPropKeyMap = ["datalabels", "hideLegend"];
+
+const availableColors = [
+  "#FF0000",
+  "#00FF00",
+  "#0000FF",
+  "#FFFF00",
+  "#FF00FF",
+  "#00FFFF",
+  "#FF4500",
+  "#8A2BE2",
+  "#32CD32",
+  "#1E90FF",
+  "#FFD700",
+  "#9932CC",
+  "#8B0000",
+  "#228B22",
+  "#008080",
+  "#FF6347",
+  "#40E0D0",
+  "#800080",
+  "#2E8B57",
+  "#FF8C00",
+  "#8B4513",
+  "#008B8B",
+  "#FF69B4",
+  "#7CFC00",
+  "#4169E1",
+  "#FF1493",
+  "#ADFF2F",
+  "#800000",
+  "#20B2AA",
+  "#F08080",
+];
+
+const COMBO_CHART_TYPE = {
+  bar: "bar",
+  line: "line",
+  stack: "bar-stack",
+  scatter: "scatter",
+};
+
+function getDataForColumn(column: ChartColumn, dataArr: DataPointsArray) {
+  const idx = _.findIndex(dataArr.columns, (colId) => column.id === colId);
+  return _.uniq(_.map(dataArr.dataValue, (row) => row[idx]));
+}
+
+function getSeriesDataForColumn(column: ChartColumn, dataArr: DataPointsArray) {
+  const idx = _.findIndex(dataArr.columns, (colId) => column.id === colId);
+  return _.uniq(_.map(dataArr.dataValue, (row) => row[idx]));
+}
+
+function getDataForColumnForSeries(
+  column: ChartColumn,
+  dataArr: DataPointsArray,
+  col: any
+) {
+  const idx = _.findIndex(dataArr.columns, (colId) => column.id === colId);
+  return dataArr.dataValue
+    .filter((item) => item[2] === col)
+    .map((row) => row[idx]);
+}
+
+function getChartDataModel(
+  configDimensions,
+  dataArr: DataPointsArray,
+  inputType
+) {
+  const xAxisColumns = configDimensions?.[0].columns ?? [];
+  const yAxisColumns = configDimensions?.[1].columns ?? [];
+  const legend = configDimensions?.[2]?.columns[0] ?? [];
```

Comment:
what happens if we have multiple legends columns?

Reply:
Added the support for handling one legend only.
Will add the logic in validation config